acceptance/bin/wait_pid.py

```diff
@@ -0,0 +1,49 @@
+#!/usr/bin/env python3
+import sys
+import os
+import time
+import platform
+import subprocess
+
+
+def wait_pid(pid):
+    max_attempts = 600  # 600 * 0.1 seconds = 1 minute
+    sleep_time = 0.1
+
+    for i in range(max_attempts):
+        # Check if we are on Windows or a Unix system.
+        if platform.system().lower() == "windows":
+            # Windows approach: use tasklist to check for the existence of the process.
+            try:
+                # Get the output of 'tasklist'
+                output = subprocess.check_output(["tasklist"], text=True)
+            except subprocess.CalledProcessError:
+                print("[wait_pid] Error retrieving tasklist. Assuming process has ended.")
+                return 0
+
+            # if the PID is not found in the list then assume the process ended.
+            if str(pid) not in output:
```

Comment:
str(pid) in output is going to have false positives, e.g. if pid=123 but output contains 1234.

Reply:
Good catch! This would have been annoying to debug. Using the `/fi` filter in tasklist itself which should take care of itself.